src/main/java/com/bootcamp/demo/service/ScooterServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Scooter;
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class ScooterServiceImpl implements ScooterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScooterServiceImpl.class);
+    private static final String COLLECTION_SCOOTERS_PATH = "bookings/databases/scooters";
+    private final Firestore db;
+
+    private ScooterServiceImpl(Firestore db) {
+        this.db = db;
+    }
+
+    @Override
+    public Scooter findScooterById(String scooterId) {
+        LOGGER.info("FIND SCOOTER BY ID - service function invoked. scooterId = {}", scooterId);
+
+        try {
+            DocumentSnapshot scooter = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooterId)
+                    .get()
+                    .get();
+
+            if (!scooter.exists()) {
+                LOGGER.warn("FIND SCOOTER BY ID - Scooter does not exist! scooterId = {}", scooterId);
+                throw new ItemNotFoundException(String.format("Scooter does not exist. scooterId = %s", scooterId));
+            }
+
+            return scooter.toObject(Scooter.class);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("FIND SCOOTER BY ID - service function. scooterId = {}", scooterId);
+            LOGGER.error(ExceptionUtils.getStackTrace(e));
```

Comment:
Highlight why logging is called twice (and not once). Also, please look at the documentation of the logger's methods since it will provide more details about how it can be used.